vnext/ReactUWP/Modules/NativeUIManager.cpp

```diff
@@ -1093,15 +1093,35 @@ void NativeUIManager::measure(
     return;
   }
 
-  // Retrieve the XAML element for the root view containing this view
-  ShadowNodeBase &nodeRoot = static_cast<ShadowNodeBase &>(shadowRoot);
-  XamlView xamlRootView = nodeRoot.GetView();
-  if (xamlRootView == nullptr) {
-    callback(args);
-    return;
+  // Traverse up the react node tree to find any windowed popups.
+  // If there are none, then we use the top-level root provided by our caller.
+  winrt::FrameworkElement feRootView = nullptr;
+  int64_t rootTag = shadowNode.m_tag;
+  while (true) {
+    auto &currNode = m_host->GetShadowNodeForTag(rootTag);
+    if (currNode.m_parent == -1)
+      break;
+    ShadowNodeBase &rootNode = static_cast<ShadowNodeBase &>(currNode);
+    if (rootNode.IsWindowed()) {
+      if (auto childView = rootNode.GetChildView()) {
```

Comment:
Do we need GetChildView implementations on shadownodebase or can we use VisualTreeHelper.GetChild to get the child of popups and flyout?

Reply:
The XAML Flyout is kinda different and I'm not sure GetChild will work, but I'll investigate.

---
In reply to: [305405289](https://github.com/microsoft/react-native-windows/pull/2781#discussion_r305405289) [](ancestors = 305405289)

Reply:
VisualTreeHelper.GetChild didn't work correctly , but I was able to use the basic functionality of NativeUIManager to tighten it up.